spec/unit/puppet-lint/plugins/check_whitespace/space_before_arrow_spec.rb

```diff
@@ -0,0 +1,126 @@
+require 'spec_helper'
+
+describe 'space_before_arrow' do
+  let(:msg) { "there should be a single space before '=>' on line %d, column %d" }
+
+  context 'with code that should not trigger any warnings' do
+    context 'resource with multiple parameters on different lines' do
+      let(:code) do
+        <<-END
+          file { 'foo':
+            foo => bar,
+            bar  => buzz,
+          }
+        END
+      end
+
+      it 'does not detect any problems' do
+        expect(problems).to be_empty
+      end
+    end
+
+    context 'resource with single param and normal spacing' do
+      let(:code) do
+        <<-END
+          file { 'foo':
+            foo => bar,
+          }
+        END
+      end
+
+      it 'does not detect any problems' do
+        expect(problems).to be_empty
+      end
+    end
+
+    context 'resource with multiple params and normal spacing' do
+      let(:code) do
+        <<-END
+          file { 'foo':
+            foo => { "bar" => "baz" },
+          }
+        END
+      end
+
+      it 'does not detect any problems' do
+        expect(problems).to be_empty
+      end
+    end
+  end
+
+  context 'resource with a single param and simple value with too much space before arrow' do
+    let(:code) do
+      <<-END
+        file { 'foo':
+          foo  => bar,
+        }
+      END
+    end
+
+    context 'with fix disabled' do
+      it 'extra space before arrow' do
```

Comment:
```suggestion
      it 'detects an extra space before arrow' do
```
Looking good. Just a small nit-pick @gerases, if you could apply this below too

Reply:
done